### PR TITLE
Articulations can be authored in Editor

### PR DESCRIPTION
**Pull request by moraaar**
## What does this PR do?

Joint limits and flags now are applied to the articulation and they also disable parent's transform changes.

Closes https://github.com/o3de/o3de/issues/15220

## How was this PR tested?

Manually authored an articulation with 4 links using hinge and prismatic joints (with limits are applied to them)

https://user-images.githubusercontent.com/27999040/229111329-c9dcded0-94fd-4e3c-aa8b-424769aa84ad.mp4

### CODE REVIEW

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 1]
```diff
@@ -254,12 +267,76 @@ namespace PhysX
...
+                inboundJoint->setJointType(physx::PxArticulationJointType::eFIX);
+                break;
+            case ArticulationJointType::Hinge:
+                inboundJoint->setJointType(physx::PxArticulationJointType::eREVOLUTE);
+                if (articulationLinkConfiguration.m_isLimited)
+                {
+                    // The lower limit should be strictly smaller than the higher limit.
+                    physx::PxArticulationLimit limits;
+                    limits.low = AZ::DegToRad(articulationLinkConfiguration.m_angularLimitNegative);
+                    limits.high = AZ::DegToRad(AZStd::max(
```
Comment (greerdv) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:287]: If you're taking the max for the high limit, should you also take the min for low? Same question for prismatic joints

**Review by greerdv — APPROVED** [review]
Looks good!

**Review by amzn-ahmadkrm — APPROVED** [review]
LGTM

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp [hunk 2]
```diff
@@ -138,17 +143,23 @@ namespace PhysX
...
+            }
 
-        if (m_articulation)
+            AZ::TransformNotificationBus::Handler::BusDisconnect();
+        }
+        else
         {
-            DestroyArticulation();
+            m_link = nullptr;
+            m_driveJoint = nullptr;
```
Comment (hultonha) [inline Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:158]: question: Just curious do these always stay as `nullptr` throughout in the case of the root articulation?

**Review by hultonha — APPROVED** [review]
LGTM! I Just have one question to confirm

File: Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp (hunk 1 above)
Comment (moraaar) [reply Gems/PhysX/Code/Source/ArticulationLinkComponent.cpp:287]: I guess I can do that as well and be a bit more user-friendly even though they entered the min-max the other way around.

**Pull request merged**